crates/solver/src/solver/score_computation.rs

```diff
@@ -0,0 +1,169 @@
+
+#[derive(Debug, Parser)]
+#[group(skip)]
+pub struct Arguments {
+    /// Parameters for the score computation for each solver.
+    /// The format is a list of semicolon separated solver parameters.
+    /// Each solver parameter is a comma separated list of parameters:
+    /// [solver name],[gas amount factor],[gas price factor],[number of orders
+    /// factor],[intercept parameter]
+    #[clap(long, env, default_value = DEFAULT_SCORE_PARAMETERS)]
+    score_parameters: ScoreParameters,
+}
+
+impl Arguments {
+    pub fn get_calculator(&self, solver: SolverType) -> Option<ScoreCalculator> {
+        self.score_parameters.0.get(&solver).cloned()
+    }
+}
+
+impl Display for Arguments {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        write!(f, "score_parameters: {:?}", self.score_parameters)
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct ScoreParameters(HashMap<SolverType, ScoreCalculator>);
+
+impl FromStr for ScoreParameters {
+    type Err = anyhow::Error;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        // parse user provided parameters (or default if none are provided)
+        let user_score_parameters = parse_calculators(s)?;
+        // parse default parameters and override them with the ones provided by the user
+        let score_parameters = parse_calculators(DEFAULT_SCORE_PARAMETERS)?
+            .into_iter()
+            .map(|(solver, default_score_parameter)| {
+                (
+                    solver,
+                    user_score_parameters
+                        .get(&solver)
+                        .cloned()
+                        .unwrap_or(default_score_parameter),
+                )
+            })
+            .collect();
+        Ok(Self(score_parameters))
+    }
+}
+
+fn parse_calculators(s: &str) -> Result<HashMap<SolverType, ScoreCalculator>> {
+    s.split(';')
+        .map(|part| {
+            let (solver, parameters) = part
+                .split_once(',')
+                .context("malformed solver score parameters")?;
+            let mut parameters = parameters.split(',');
+            let a = parameters
+                .next()
+                .context("missing a parameter for score")?
+                .parse()?;
+            let b = parameters
+                .next()
+                .context("missing b parameter for score")?
+                .parse()?;
+            let c = parameters
+                .next()
+                .context("missing c parameter for score")?
+                .parse()?;
+            let x = parameters
+                .next()
+                .context("missing x parameter for score")?
+                .parse()?;
+            Ok((
+                SolverType::from_str(solver, true).map_err(|message| anyhow::anyhow!(message))?,
+                ScoreCalculator::new(a, b, c, x),
+            ))
+        })
+        .collect::<Result<HashMap<_, _>>>()
+}
+
+#[cfg(test)]
+mod tests {
```

Comment:
Maybe there should be at least one test to make sure the calculation is right. With an example from the solver team to check against.

Reply:
YES! Was planning to test against one. I committed a test and now waiting for approval from @harisang

Reply:
Added test and confirmed the calculation with solver team.